compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -640,6 +640,17 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // extra info for `c"str"` before 2021 edition or `c "str"` in all editions
+        if self.prev_token.is_ident_named(Symbol::intern("c"))
+            && let TokenKind::Literal(token::Lit { kind: token::Str | token::StrRaw(..), .. }) =
+                &self.token.kind
+        {
+            err.note("you may be trying to declare a CStr literal");
+            err.note("`CStr` literals require edition 2021");
```

Comment:
Backticks in diagnostics are for quoting code and user-written text, not for names.
```suggestion
            err.note("`c-string literals require at least edition 2021");
```